servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/ConnectionAcceptorFactoryAppender.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2023 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.transport.api;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A special {@link ConnectionAcceptorFactory} which allows to append two factories together.
+ * <p>
+ * Note that both factories' execution strategies are merged together and that result is used when
+ * {@link #requiredOffloads()} is called.
+ */
+public final class ConnectionAcceptorFactoryAppender implements ConnectionAcceptorFactory {
```

Comment:
By exposing a new public class we don't reduce number of ways how to append ConnectionAcceptorFactories. The expectation was that `HttpServerBuilder#appendConnectionAcceptorFilter(ConnectionAcceptorFactory)` will be the only way to append acceptors, working in a way similar to `appendServiceFilter`